Support LLVM 16

### Description

Allow HOOMD-blue to compile against LLVM 16.

### Proposed solution

Solution unkown at this time.

### Additional context

The first compile error is:
```
/Users/joaander/devel/hoomd/hoomd/hpmc/ClangCompiler.cc:70:11: error: no type named 'initializeInstrumentation' in namespace 'llvm'
    llvm::initializeInstrumentation(Registry);
```

--- a/hoomd/hpmc/ClangCompiler.cc
+++ b/hoomd/hpmc/ClangCompiler.cc
@@ -67,7 +67,9 @@ ClangCompiler::ClangCompiler()
     llvm::initializeAnalysis(Registry);
     llvm::initializeTransformUtils(Registry);
     llvm::initializeInstCombine(Registry);
+#if LLVM_VERSION_MAJOR < 16
     llvm::initializeInstrumentation(Registry);
+#endif
     llvm::initializeTarget(Registry);
     }
 

